packages/cli/src/commands/source/push/delete.ts

```diff
@@ -0,0 +1,165 @@
+import {Command, flags} from '@oclif/command';
+import {Source} from '@coveo/push-api-client';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import dedent from 'ts-dedent';
+import {
+  buildAnalyticsFailureHook,
+  buildAnalyticsSuccessHook,
+} from '../../../hooks/analytics/analytics';
+import {green, red} from 'chalk';
+// eslint-disable-next-line node/no-extraneous-import
+import {AxiosResponse} from 'axios';
+
+interface ErrorFromAPI {
+  response: {
+    status: number;
+    data: {
+      errorCode: string;
+      message: string;
+    };
+  };
+}
+
+export default class SourcePushDelete extends Command {
+  public static description =
+    'Delete one or multiple documents in a given push source. See https://docs.coveo.com/en/171 and https://docs.coveo.com/en/131';
+
+  public static flags = {
+    deleteOlderThan: flags.string({
+      char: 'd',
+      exclusive: ['delete'],
+      description:
+        'Delete old items, using either an ISO 8601 date or a Unix timestamp',
+      helpValue: '2000-01-01T00:00:00-06:00 [ OR ] 946702800',
+    }),
+    delete: flags.string({
+      char: 'x',
+      multiple: true,
+      description:
+        'Document URI or identfier to delete. Can be repeated. If you want to delete a large batch of documents, use source:push:batch command instead.',
+    }),
+    deleteChildren: flags.boolean({
+      char: 'c',
+      default: true,
+      description:
+        'Specify if children document should also be deleted. Default to `true`.',
+    }),
+  };
+
+  public static args = [
+    {
+      name: 'sourceId',
+      required: true,
+      description:
+        'The identifier of the source on which to perform the delete operation. See source:push:list to obtain the identifier.',
+    },
+  ];
+
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    const {flags} = this.parse(SourcePushDelete);
+    if (!flags.deleteOlderThan && !flags.delete) {
+      this.error(
+        'You must provide either --delete= or --deleteOlderThan=. See source:push:delete --help for more information.'
+      );
+    }
+    const cfg = await new AuthenticatedClient().cfg.get();
+    const source = new Source(cfg.accessToken!, cfg.organization);
+
+    if (flags.deleteOlderThan) {
+      this.doDeletionOlderThan(source);
+      return;
+    }
+
+    if (flags.delete) {
+      if (this.isNumberOfDeletionTooLarge()) {
+        return;
+      }
+
+      await this.doDeletionDocumentURI(source);
+    }
+    await this.config.runHook('analytics', buildAnalyticsSuccessHook(this, {}));
+  }
+
+  private isNumberOfDeletionTooLarge() {
+    if (this.flags.delete.length > 20) {
+      this.warn(
+        dedent(
+          'To delete large batch of documents, use source:push:batch command instead.'
+        )
+      );
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
I would put the warn in the `run` because the warning goes against the name of the function IMO: The `isSomethingSomething` imo should not alter anything but just run a check and return true or false